data_transfer/jobs/shared.py

```diff
@@ -26,3 +40,57 @@ def batch_upload_data() -> None:
                 record.is_uploaded = True
                 update_record(record)
             dmpy.rm_local_data(zip_path)
+
+
+def prepare_data_folders(device_type: DeviceType) -> None:
+    """
+    Checks folders present in 'data/upload/ are finished
+    and moves them into the upload folder in the format:
+
+        DEVICEID-PATIENTID-STARTWEAR-ENDWEAR
+    """
+    not_uploaded = records_not_uploaded(device_type)
+
+    grouped: dict = {}
+    # transform the result to {PatientID1-DeviceID1: [{Record1}, ... {Record2}], ... }
+    [
+        grouped.setdefault(f"{r.patient_id}/{r.device_id}", []).append(r)
+        for r in not_uploaded
+    ]
+
+    # filter sets which have any record with 'is_processed' == False
+    # 'is_processed' == False catches the 'False' for any preceding task as well
+    to_upload = {k: v for k, v in grouped.items() if all([r.is_processed for r in v])}
+
+    for patient_device in to_upload.keys():
+        max_data, min_data = min_max_data_wear_times(to_upload[patient_device])
+
+        start_data = max_data.strftime("%Y%m%d")
+        end_data = min_data.strftime("%Y%m%d")
+
+        source = config.storage_vol / patient_device
+        destination = (
+            config.upload_folder
+            / device_type.name
+            / f"{patient_device.replace('-','').replace('/','-')}-{start_data}-{end_data}"
+        )
+
+        if not destination.exists():
+            destination.mkdir(parents=True)
+
+        source.rename(destination)
+
+        # don't forget the -meta.json file! Each record in the list has a reference to it
+        meta_file_name = f"{to_upload[patient_device][0].manufacturer_ref}-meta.json"
+        meta_file_path = config.storage_vol / meta_file_name
+        meta_file_path.rename(destination / meta_file_name)
```

Comment:
This could be removed if `-meta.json` files are in the `source` folder. Same is true for all pre-processed files.